### PR TITLE
Check instalation source on Andriod and open github page instead of google play store if the instalation source is not google play

### PR DESCRIPTION
**Pull request by ademar111190**
Related https://github.com/breez/breezmobile/issues/413

If the app is installed from Google Play, it opens Google Play as it is today, if the installation source is somewhere else, it opens the release page on GitHub

Also breaking the method into smaller methods and adding logs

How it looks like:

https://github.com/breez/breezmobile/assets/1225438/788e24cf-33e1-48a9-b535-07a028dfcd5b

### CODE REVIEW

File: lib/handlers/check_version_handler.dart [hunk 1]
```diff
@@ -4,80 +4,132 @@ import 'package:breez/bloc/blocs_provider.dart';
...
+  _log.info("_handleBadVersion");
+  final texts = context.texts();
+
+  showFlushbar(
+    context,
+    buttonText: texts.handler_check_version_action_update,
+    onDismiss: () {
+      _log.info("onDismiss $defaultTargetPlatform");
+      if (defaultTargetPlatform == TargetPlatform.iOS) {
+        launchUrlString('https://testflight.apple.com/join/wPju2Du7');
```
Comment (ubbabeck) [inline lib/handlers/check_version_handler.dart:102]: Why are we not directing ios users to app store?

**Review by ubbabeck — APPROVED** [review]
lgmt

File: lib/handlers/check_version_handler.dart (hunk 1 above)
Comment (ademar111190) [reply lib/handlers/check_version_handler.dart:102]: Good question; do you have context on this @erdemyerebasmaz ?

Comment (erdemyerebasmaz) [reply lib/handlers/check_version_handler.dart:102]: @ademar111190 No, but why not use `Source.IS_INSTALLED_FROM_APP_STORE` & `Source.IS_INSTALLED_FROM_TEST_FLIGHT` for redirects on iOS?

Comment (ademar111190) [reply lib/handlers/check_version_handler.dart:102]: Good call out; I'll do it.

**Review by erdemyerebasmaz — APPROVED** [review]

**Pull request merged**